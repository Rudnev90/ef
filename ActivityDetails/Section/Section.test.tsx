import * as React from 'react';

import { identity } from 'fp-ts/lib/function';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import { Section, SectionProps } from '.';

const createJson = (props: SectionProps) => renderer.create(
  <Section {...props} />,
  { createNodeMock: identity }, // YB: See https://github.com/facebook/react/issues/7740#issuecomment-247335106
).toJSON();

describe('Section', () => {
  it.concurrent('Should render according to the snapshot', async () => {
    const output: ReactTestRendererJSON = createJson({ icon: 'phone', classes: {}, theme: null });
    expect(output).toMatchSnapshot();
  });

  it.concurrent('Should render with direaction column', async () => {
    const output: ReactTestRendererJSON = createJson({ icon: 'phone', classes: {}, theme: null, direction: 'column' });
    expect(output).toMatchSnapshot();
  });
});
