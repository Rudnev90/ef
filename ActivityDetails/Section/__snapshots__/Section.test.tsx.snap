// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-wrap-xs-nowrap-7 MuiGrid-align-items-xs-center-9"
>
  <div
    className="MuiGrid-item-2"
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-98"
    >
      phone
    </span>
  </div>
  <div
    className="MuiGrid-item-2"
  />
</div>
`;

exports[` 2`] = `
<div
  className="MuiGrid-container-1 MuiGrid-direction-xs-column-4 MuiGrid-wrap-xs-nowrap-7 MuiGrid-align-items-xs-center-9"
>
  <div
    className="MuiGrid-item-2"
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-98"
    >
      phone
    </span>
  </div>
  <div
    className="MuiGrid-item-2"
  />
</div>
`;
